dashboard/app/models/script.rb

```diff
@@ -144,6 +144,7 @@ def generate_plc_objects
     project_sharing
     curriculum_umbrella
     tts
+    is_course
```

Comment:
Should is_course be a column or a property? It seems like we might want to be able to search for all the scripts that are courses easily? Does content root allow us to do that so we aren't worried about using this?

Reply:
this field seems like redundant state. how about this instead:
```
def is_course
  !!course_version
end
```

Reply:
Oh, I remember why I did this now. This field is only really needed to determine whether we should create a CourseVersion to associate with this Script in the seeding process. (Looking at presence of `version_year` doesn't work there because scripts which aren't content roots have those too). 

This is the easiest way to get that piece of information from the .script file into the seeding logic. I could explicitly avoid storing it in the DB by leaving it off the properties here, though it'd still have to be included in `build_property_hash` (or we'd have to refactor how this works.) Not sure if that's any better.

Which do you prefer? Any other ideas?

Reply:
Good points Winter. I think the redundant state is ok as long as it serves a purpose, which it sounds like it does. So either option seems fine, so in that sense it doesn't seem worth redoing. Another idea (since you asked 😉  ) would be to add a validation enforcing that `!!course_version == is_course`.

Reply:
I feel like this validation might cause problems when we update the Script object from the DSL properties, which happens before we would create the corresponding CourseVersion objects.

I could believe there's a good way to refactor things to make this all better but it's not obvious to me right now. I'm open to continuing to iterate on this, but I'll leave this as-is for this PR.